cocotb/share/include/cocotb_bfm_api.h

```diff
@@ -0,0 +1,118 @@
+/******************************************************************************
+ * Copyright cocotb contributors
+ * Licensed under the Revised BSD License, see LICENSE for details.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ******************************************************************************/
+#ifndef INCLUDED_COCOTB_BFM_API_H
+#define INCLUDED_COCOTB_BFM_API_H
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Callback function type used to notify
+// implementation when a message is available
+typedef void (*cocotb_bfm_notify_f)(void *);
+
+int cocotb_bfm_register(
+        const char                *type_name,
+        const char                *inst_name,
+        const char                *cls_name,
+        cocotb_bfm_notify_f        notify_f,
+        void                    *notify_data);
```

Comment:
Nit: some small misalignment (from the tab cleanup?)

Reply:
Done